client/Components/Include/MyRecipes.jsx

```diff
@@ -70,12 +81,122 @@ class MyRecipes extends Component {
     })
   }
 
-  onSubmit() {
+  uploadImage(event) {
+    event.preventDefault();
+    let name = event.target.files[0];
+    let file_reader = new FileReader();
+    if (name) {
+      file_reader.onload = () => {
+        const newImage = new Image();
+        newImage.src = file_reader.result;
+        newImage.onload = () => {
+          this.setState({
+            imageHeight: newImage.height,
+            imageWidth: newImage.width,
+            image: name
+          });
+        }
+      }
+    }
+    file_reader.readAsDataURL(name);
+  }
+
+  onSubmit(event) {
+    event.preventDefault();
+    const recipeName = this.state.recipeName
+    const details = this.state.details
+    const ingredient = this.state.ingredient
+    if(recipeName.trim() === '' 
+    || details.trim() === '' 
+    || ingredient.trim() === '') {
+      toastr.options = {
```

Comment:
You are constantly reusing this configuration. It would be better extracting it into a helper. Try not to violate the DRY principle. @Ruqoyah

Reply:
Noted, thanks